nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/controller/ControllerFacade.java

```diff
@@ -1389,104 +1420,119 @@ private ProvenanceEventDTO createProvenanceEventDto(final ProvenanceEventRecord
         // sets the component details if it can find the component still in the flow
         setComponentDetails(dto);
 
-        // only include all details if not summarizing
-        if (!summarize) {
-            // convert the attributes
-            final Comparator<AttributeDTO> attributeComparator = new Comparator<AttributeDTO>() {
-                @Override
-                public int compare(AttributeDTO a1, AttributeDTO a2) {
-                    return Collator.getInstance(Locale.US).compare(a1.getName(), a2.getName());
-                }
-            };
+//        try {
+//            AuthorizationResult result = flowController.checkConnectableAuthorization(event.getComponentId());
+        AuthorizationResult result = checkConnectableAuthorization(event.getComponentId());
+            if (Result.Denied.equals(result.getResult())) {
+                dto.setComponentType("Processor"); // is this always a Processor?
+                dto.setComponentName(dto.getComponentId());
+                dto.setEventType("UNKNOWN");
```

Comment:
Do you think that we need to redact the event type when the user does not have permissions to the component policy? I would have considered this field under the new provenance event policy.

Reply:
If we choose to _not_ redact event type, that makes life easier. Currently, it displays "UNKNOWN" in the table (when 'view provenance' is enabled and 'view the component' is not). But, the event type IS diplayed in the lineage graph. We need to get to consistency one way or the other on this. I'm leaning towards allowing the event type info to be visible since this is a characteristic of provenance (i.e. 'view provenance') and not a characteristic of 'view the component'.

Reply:
Yes, I agree. The event type should be controlled by the new provenance event policy. It is not controlled by the component policy that protects the component name and component type.